### DISCUSSION TITLE
Class içindeki metodu class için nasıl kullabilirim?

### DISCUSSION THREAD

**Discussion by BunyaminAkdeniz** [19/classroom#12] [Ruby] [answered]
Merhaba hocam. Class içinde metod yazıp onları class için kullanmak istiyorum . Örneğin 
```ruby
class HesapMakinesi
  def mod_3(num)
    num % 3
  end

  def self.hesapla(num)
    self.mod_3(num)
  end
end


```
 HesapMakinesi.hesapla 3 # => 0 vermesini istiyorum fakat no method alıyorum. mod_3 metodunu nerede ne şekilde tanımlamam gerekiyor ?

**Comment by roktas** [comment, marked as answer]
`mod_3` metodu da bir sınıf metodu olmalı, `self`'e dikkat.  Ruby'de aynı odadaki metotlar birbirini çağırabilir ancak.  Oda?  `self`

```ruby
class HesapMakinesi
  def self.mod_3(num)
    num % 3
  end

  def self.hesapla(num)
    mod_3(num) # ayrıca başa self koyman gerekmiyor, "needless self"
  end
end
```

P.S. Bu örnekte her hangi bir nesne "instantaniate" etmediğinden `class` yerine `module` kullanman şu haliyle daha doğru.  Kodun nereye evrileceğine bağlı.  Bu tür kodlar genellikle modülden class'a doğru evrileceğinden geleceği dikkate alarak `class` ile de gidebilirsin, ama o durumda "sınıf metotları" yerine "nesne metotları" tercih edilmeli.  Buna NYP dersinde ayrıntılı gireceğiz, şimdilik kısa geçiyorum.

**Reply by BunyaminAkdeniz** [reply]
Çok teşekkür ederim hocam, bu kısmı iyi anladım
